api/controllers/Sections.php

```diff
@@ -179,8 +179,8 @@ public function POST () {
 		if(isset($this->_params->masterSection)) {
 			$masterSection = $this->Sections->fetch_section ("id", $this->_params->masterSection);
 			// checks
-			if(sizeof($masterSection)==0)				{ $this->Response->throw_exception(400, 'Invalid masterSection id '.$this->_params->masterSection); }
-			elseif($masterSection->masterSection!="0")	{ $this->Response->throw_exception(400, 'Only 1 level of nesting is permitted for sections');  }
+			if(count(get_object_vars($masterSection))==0)	{ $this->Response->throw_exception(400, 'Invalid masterSection id '.$this->_params->masterSection); }
```

Comment:
fetch_section() returns an object or false.

Can you change `if(count(get_object_vars($masterSection))==0)` to `if (!is_object($masterSection)) { ... }` 

```php
Sections::fetch_section

fetches section by specified method

public function fetch_section($method, $value) { }

@param string $method

@param mixed $value

@return object|bool

@access  public
```

Reply:
Sure, that's done now and committed.